app/features/receive/receive-cashu.tsx

```diff
@@ -116,49 +119,56 @@ function CashuRequestQRItem({ account, amount }: CashuRequestQRProps) {
 }
 
 type MintQuoteProps = {
-  account: Account & { type: 'cashu' };
+  account: CashuAccount;
   amount: Money;
   isVisible: boolean;
 };
 
 function MintQuoteItem({ account, amount, isVisible }: MintQuoteProps) {
   const [, copyToClipboard] = useCopyToClipboard();
-  const { mintQuote, createQuote, fetchError, checkError, isLoading } =
-    useMintQuote({
-      account,
-      amount,
-    });
   const { toast } = useToast();
+  const {
+    mutate: createQuote,
+    data: createdQuote,
+    isPending,
+    error,
+  } = useCreateCashuReceiveQuote();
 
-  useEffect(() => {
-    if (checkError) {
+  const { quote, status } = useCashuReceiveQuote({
+    quoteId: createdQuote?.id,
+    onExpired: () => {
       toast({
-        title: 'Error',
-        description: checkError,
-        variant: 'destructive',
+        title: 'Quote expired',
+        description: 'Please create a new quote.',
       });
-    }
-  }, [checkError, toast]);
+    },
+    onPaid: () => {
+      toast({
+        title: 'Quote paid',
+        description: 'Please create a new quote.',
+      });
+    },
+  });
 
   useEffect(() => {
-    if (isVisible && !mintQuote) {
-      createQuote();
+    if (isVisible && !quote) {
+      createQuote({ account, amount });
     }
-  }, [isVisible, createQuote, mintQuote]);
+  }, [isVisible, createQuote, quote, account, amount]);
 
   return (
     <QRCarouselItem
-      value={mintQuote?.request}
-      description="Scan with any Lightning wallet."
-      error={fetchError || checkError}
-      isLoading={isLoading}
+      value={quote?.paymentRequest}
+      description={`Scan with any Lightning wallet. Status: ${status}`}
```

Comment:
@gudnuf I added status to this message just temporarily so we can see how it is changing but you can remove that when you will be working on proper handling of expired and paid states in ui